src/Symfony/Component/Console/Application.php

```diff
@@ -271,25 +289,22 @@ public function doRun(InputInterface $input, OutputInterface $output)
                     $e = $event->getError();
                 }
 
-                throw $e;
-            }
-
-            $alternative = $alternatives[0];
-
-            $style = new SymfonyStyle($input, $output);
-            $style->block(sprintf("\nCommand \"%s\" is not defined.\n", $name), null, 'error');
-            if (!$style->confirm(sprintf('Do you want to run "%s" instead? ', $alternative), false)) {
-                if (null !== $this->dispatcher) {
-                    $event = new ConsoleErrorEvent($input, $output, $e);
-                    $this->dispatcher->dispatch($event, ConsoleEvents::ERROR);
-
-                    return $event->getExitCode();
+                try {
+                    if ($e instanceof CommandNotFoundException && $namespace = $this->findNamespace($name)) {
+                        $helper = new DescriptorHelper();
+                        $helper->describe($output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output, $this, [
+                            'format' => 'txt',
+                            'raw_text' => false,
+                            'namespace' => $namespace,
+                            'short' => false,
+                        ]);
+
+                        return $event->getExitCode();
```

Comment:
Are we sure that `$event` always exists here?

Reply:
Great catch, we are not :)

I'm slightly wondering why Psalm isn't screaming about this, but I'm almost certain this may not be defined (when `$this->dispatcher` is null)